src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC } from 'react';
+import { Icon as IconifyIcon } from '@iconify/react';
+
+interface Props {
+
+  /**
+   * Name of icon in format provider:icon-name.
+   */
+  name: string;
+
+  /**
+   * Style object.
+   */
+  style?: React.CSSProperties;
+
+  /**
+   * Class name.
+   */
+  className?: string;
+
+  /**
+   * Size in pixels.
+   */
+  size?: number | typeof stringSizes[number];
+}
+
+const defaultProps = {
+  style: undefined,
+  className: undefined,
+  size: 'md' as const,
+};
+
+const stringSizes = ['md', 'sm', 'xs'] as const;
+
+/**
+ * TODO: Icon.
+ */
+export const Icon: FC<Props> = ({ name, style, className, size }) => {
```

Comment:
I don't think this `defaultProps` object is necessary here because unrequired props are `undefined` by default. To define default props I find it easier and more convinient to do it like this:
`export const Icon: FC<Props> = ({ name, style, className, size = 'md' }) => {`